packages/store/src/decorators/action.ts

```diff
@@ -1,16 +1,68 @@
 import { ɵActionOptions, ɵensureStoreMetadata } from '@ngxs/store/internals';
 
-import { ActionType } from '../actions/symbols';
+import { ActionDef, ActionType } from '../actions/symbols';
 import { throwActionDecoratorError } from '../configs/messages.config';
+import { StateContext } from '../symbols';
 
 /**
- * Decorates a method with a action information.
+ * Given an action class, returns its payload.
  */
-export function Action(
-  actions: ActionType | ActionType[],
+type ActionToPayload<Action extends ActionType> =
+  Action extends ActionDef<any, infer ActionPayload> ? ActionPayload : never;
+
+/**
+ * Given a list of action classes, returns the union of their payloads.
+ */
+type ActionsToPayload<Actions extends readonly ActionType[]> = {
+  [K in keyof Actions]: Actions[K] extends ActionDef<any, infer ActionPayload>
+    ? ActionPayload
+    : never;
```

Comment:
Would this work here?
```suggestion
  [K in keyof Actions]: ActionToPayload<Actions[K]>;
```